app/controllers/application_controller/compare.rb

```diff
@@ -420,7 +420,7 @@ def drift_same
   # AJAX driven routine to check for changes in ANY field on the form
   def sections_field_changed
     @keep_compare = true
-    @explorer = %w[VMs Templates].include?(session[:db_title])
+    @explorer = %w[VMs Templates].include?(session[:db_title]) && @display.nil?
```

Comment:
Fixes the issue, but I think the condition is wrong anyway.

What we usually do to determine explorer:

* `if BASE_MODEL_EXPLORER_CLASSES.include?(cls)`
* `if request.xml_http_request?`
* `if @edit[:explorer]`

Can any of those work instead?

Reply:
The old logic of "VM" or "Template" is in title => explorer, is wrong, so this is fixing a wrong thing by adding stuff instead of replacing to logic to make sense.

Perhaps the place which renders the template should be setting `@edit[:explorer]` and then we can get it from session.

Reply:
I've tried the second two of them, before I made the change in this PR, of course.

`if @edit[:explorer]` did not work, we need `@edit` to be set, for this case - and this is not always done even when we are in an explorer screen. Even when I've set `@edit[:explorer]` to `true` at the beginning of comparing items (after user clicked on Compare action), later right after calling `sections_field_changed` `@edit` was again `nil`, the information 'got lost'.
And `if request.xml_http_request?` returned `true` even when we weren't in explorer screen (this happens to me also for other places and controllers). So I've

Reply:
OK, merging as is then, sounds like a future refactoring :).

Reply:
Also checking some classes was not enough to decide if we are in a explorer screen or not, because of nested lists which are mostly in non-explorer screens.